### PR TITLE
Add organization gallery

### PR DESCRIPTION
**Pull request by Gula1507**
Feature: App shows all organizations

- Added backend to fetch organizations.
- Implemented global exception handler.
- Created frontend to display organizations.
- Added styling.

### CODE REVIEW

File: backend/src/test/java/de/neuefische/backend/service/OrganizationServiceTest.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    OrganizationRepository organizationRepository = mock(OrganizationRepository.class);
+    OrganizationService organizationService = new OrganizationService(organizationRepository);
+
+    @Test
+    void getAllOrganizations_returnsEmptyList_whenRepositoryIsEmpty() {
+        List<Organization> expected = List.of();
+        when(organizationRepository.findAll()).thenReturn((List.of()));
+        List<Organization> actual = organizationService.getAllOrganizations();
+        Assertions.assertEquals(expected, actual);
+        verify(organizationRepository).findAll();
```
Comment (Ropold) [inline backend/src/test/java/de/neuefische/backend/service/OrganizationServiceTest.java:24]:
```suggestion
        verify(organizationRepository).findAll();
        Assertions.assertEquals(expected, actual);
```

File: frontend/src/components/OrganizationCard.tsx [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+type OrganizationPageProps = {
+    organization: Organization
+}
+
+function OrganizationCard(props: OrganizationPageProps) {
+    return (
+        <div className="card">
+            <h2 className="card-title">{props.organization.name}</h2>
+            <a className="card-homepage" href={props.organization.homepage}>{props.organization.homepage}</a>
```
Comment (Ropold) [inline frontend/src/components/OrganizationCard.tsx:12]: dont use href.
use Link or usenavigate instead.

**Review by Ropold — APPROVED** [review]
Reviewed to the best of my knowledge

File: backend/src/main/java/de/neuefische/backend/model/Organization.java [hunk 3]
```diff
@@ -0,0 +1,8 @@
+package de.neuefische.backend.model;
+
+public record Organization(
+        int id,
```
[the remark this answers is not in the document]
Comment (Gula1507) [reply backend/src/main/java/de/neuefische/backend/model/Organization.java:4]: I chose int to ensure better compatibility with external APIs that use numeric id. May be if I decide to use UUIDs i will use String for the id.

**Pull request merged**